ui/app/ducks/metamask/metamask.js

```diff
@@ -154,9 +154,26 @@ function reduceMetamask (state, action) {
       return newState
 
     case actions.SET_SELECTED_TOKEN:
-      return extend(metamaskState, {
+      newState = extend(metamaskState, {
         selectedTokenAddress: action.value,
       })
+      const newSend = extend(metamaskState.send)
+
+      if (metamaskState.send.editingTransactionId && !action.value) {
```

Comment:
What about the case when a user has ETH + two or more tokens in the dropdown. Do we need to modify `unapprovedTxs` in state if, while editing a token transaction, a user switches it to a different token?

Reply:
This part is handled in `ui/app/pages/send/send.component.js:L169`. Would love a manual test from you to see if it works. I tested it with multiple tokens locally many times.